### ISSUE TITLE
{"Could not load file or assembly 'BouncyCastle.Crypto, Version=1.7.4137.9688, Culture=neutral, PublicKeyToken=a4292a325f69b123' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)":"BouncyCastle.Crypto, Version=1.7.4137.9688, Culture=neutral, PublicKeyToken=a4292a325f69b123"}

### ISSUE THREAD

**Issue by danielcamargo** [bitpay/csharp-bitpay-client#20]

**Comment by TopperDEL** [comment]
Any news on this? I get the same error and can't find a solution...

**Comment by FlYnN91** [comment]
I found the fix for this. Bitpay's C# API has a dependency of BitCoinSharp library which depends on BouncyCastle.Crypto. Bitpay's C# API also links to BouncyCastle.Crypto but a never version. What this error is saying, is that the version of BouncyCastle.Crypto you are using is not what the BitCoinSharp is expecting. What I did:
- Install BitCoinSharp via nuget in VisualStudio. BouncyCastle.Crypto gets installed as dependency too.
- Clone the BitPay repo and add the project to the VisualStudio solution.
- Go to the folder where nuget installed packages are stored
- Copy the BouncyCastle.Crypto.dll from the nuget to BitPay project's redist folder.
- Rebuild BitPay project

Hope this helps.

**Comment by ajp8164** [comment]
@FlYnN91 is correct, the BitCoinSharp library is built with BouncyCastle 1.7.4137.9688.  You must include this version of BC with your build.

**Comment by quappinger** [comment]
I have included BouncyCastle 1.7.4137.9688 but now the build fails and the error message tells me, that the build needs "BouncyCastle.Crypto, Version=1.7.4114.6375, Culture=neutral, PublicKeyToken=0e99375e54769942". This is confusing. Also tried to install BC 1.8.1 but also had no success.

**Comment by TikyConte** [comment]
I have same problem as quappinger
no way how to fix this problem...

**Comment by GaaTY** [comment]
This issue was closed but the problem persists. I followed the steps to the letter but to no avail.

1. Create new Visual Studio C# Project

2. Install BitPay Version 2.0.0 from NuGet Package Manager Console within Visual Studio

3. Follow step number 1 in GUIDE.md:

`// Let the SDK store the private key on the clients local file system.
BitPay bitpay = new BitPay();`

4. Debug App, get error "Could not load file or assembly 'BitCoinSharp, Version=**0.2.199.60'**

5. Lookup BitCoinSharp on NuGet, version 0.2.199.60 doesn't exist.

6 Install BitCoinSharp through NuGet Package Manager from Visual Studio - version 0.2.166.49 gets installed (which is the latest version available)

7. Reference to BouncyCastle.Crypto gets installed, version 1.7.4137.9688 (correct version as mentioned above).

8 Debug App, get error: "Could not load file or assembly 'BouncyCastle.Crypto, Version=**1.7.4114.6375**"

So, where am I going wrong?
